listing-rewards/program/src/rewardable_collection/create/mod.rs

```diff
@@ -0,0 +1,59 @@
+use anchor_lang::{prelude::*, AnchorDeserialize};
+use mpl_auction_house::{self, constants::PREFIX, AuctionHouse};
+
+use crate::{
+    assertions::*,
+    constants::{REWARDABLE_COLLECTION, REWARD_CENTER},
+    errors::ListingRewardsError,
+    state::{RewardCenter, RewardableCollection},
+};
+
+#[derive(AnchorSerialize, AnchorDeserialize, Debug)]
+pub struct CreateRewardableCollectionParams {
+    pub collection: Pubkey,
+}
+
+/// Accounts for the [`create_rewardable_collection` handler](listing_rewards/fn.create_rewardable_collection.html).
+#[derive(Accounts, Clone)]
+#[instruction(rewardable_collection_params: CreateRewardableCollectionParams)]
+pub struct CreateRewardableCollection<'info> {
+    /// The wallet of collection maintainer. Either the auction house authority or collection oracle.
+    #[account(mut)]
+    pub wallet: Signer<'info>,
+
+    /// The auctioneer program PDA running this auction.
+    #[account(init, payer = wallet, space = RewardableCollection::size(), seeds = [REWARDABLE_COLLECTION.as_bytes(), reward_center.key().as_ref(), rewardable_collection_params.collection.as_ref()], bump)]
```

Comment:
We likely need to use init_if_needed to be able to toggle between an active and inactive rewardable_account.

nothing here is susceptible to a reinitialization attack.

Reply:
Couldn't we check this over the client side through getAccountInfo(), and just get the state if it's already initialized, if not, init it? I feel init_if_needed is a bit unnecessary, as we can have sufficient client checks if I'm not wrong. @kespinola

Reply:
Done :) although, I'm quite concerned about this feature though, as we can indeed have some client checks to either fetch already initialized state and only then init if not.